src/main/java/emissary/place/ComparisonPlace.java

```diff
@@ -0,0 +1,142 @@
+package emissary.place;
+
+import emissary.core.DiffCheckConfiguration;
+import emissary.core.IBaseDataObject;
+import emissary.core.IBaseDataObjectHelper;
+import emissary.core.ResourceException;
+import emissary.util.PlaceComparisonHelper;
+
+import org.apache.commons.lang3.Validate;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This place takes two other places, givens them exactly the same IBDO and compares the output.
+ * <p>
+ * The configuration file must contain the five properties defined by the five constants in this class.
+ */
+public class ComparisonPlace extends ServiceProviderPlace {
+    /**
+     * The full pathname for the class for the first class.
+     */
+    public static final String PLACE_A_CLASSNAME = "PLACE_A_CLASSNAME";
+    /**
+     * The full pathname for the configuration for the first class.
+     */
+    public static final String PLACE_A_CONFIGNAME = "PLACE_A_CONFIGNAME";
+    /**
+     * The full pathname for the class for the second class.
+     */
+    public static final String PLACE_B_CLASSNAME = "PLACE_B_CLASSNAME";
+    /**
+     * The full pathname for the configuration for the second class.
+     */
+    public static final String PLACE_B_CONFIGNAME = "PLACE_B_CONFIGNAME";
+    /**
+     * An identifier to be added to the log message.
+     */
+    public static final String LOGGING_IDENTIFIER = "LOGGING_IDENTIFIER";
+
+    private static final String NOT_NULL_STRING = " == null not allowed!";
+    private static final String PROCESS_PROCESSHD_MSG = "Mixed process/processHeavyDuty not allowed!";
+
+    private final ServiceProviderPlace placeA;
+    private final ServiceProviderPlace placeB;
+    private final String loggingIdentifier;
+
+    public ComparisonPlace(String configFile, String theDir, String thePlaceLocation) throws IOException {
+        super(configFile, theDir, thePlaceLocation);
+
+        final String placeAClassName = configG.findStringEntry(PLACE_A_CLASSNAME);
+        final String placeAConfigName = configG.findStringEntry(PLACE_A_CONFIGNAME);
+        final String placeBClassName = configG.findStringEntry(PLACE_B_CLASSNAME);
+        final String placeBConfigName = configG.findStringEntry(PLACE_B_CONFIGNAME);
+
+        loggingIdentifier = configG.findStringEntry(LOGGING_IDENTIFIER);
+
+        Validate.notNull(placeAClassName, PLACE_A_CLASSNAME + NOT_NULL_STRING);
+        Validate.notNull(placeAConfigName, PLACE_A_CONFIGNAME + NOT_NULL_STRING);
+        Validate.notNull(placeBClassName, PLACE_B_CLASSNAME + NOT_NULL_STRING);
+        Validate.notNull(placeBConfigName, PLACE_B_CONFIGNAME + NOT_NULL_STRING);
+        Validate.notNull(loggingIdentifier, LOGGING_IDENTIFIER + NOT_NULL_STRING);
```

Comment:
These validation messages don't really tell the dev how to fix the misconfiguration. "PLACE_A_CLASSNAME == null not allowed!"  is not as instructive as "No configuration entry found with key PLACE_A_CLASSNAME"